### PR TITLE
Add in a new setup option for asynchronously loading API key

### PR DESCRIPTION
**Pull request by elliottminns**
The setup option of `api_key_cmd` is great, but it's performed synchronously. As I use a password store backed by a Yubikey, this can cause a significant delay when opening up Neovim.

In order to solve this, I've added a new setup option of `async_api_key_cmd` which uses plenary.job to asynchronously load the API key. For me, this improves the performance of opening up neovim and makes the plugin usable again. 

There are some caveats with this, however.

You can only specify a command that follows the following format:

```
<cmd> <args>
```

Anything with piping, won't work. i.e.

```
<cmd> <args> | <cmd2>
```

Therefore, I opted to add a new setup configuration value, to ensure this does not break existing configurations. There may be a way to improve on this so that any command can be performed by the plenary job module.

### CODE REVIEW

**Comment by jackMort** [comment]
Great work, thanks! Could you just fix the styling errors to pass checks?

File: README.md [hunk 1]
```diff
@@ -54,6 +54,7 @@ use({
 ```lua
 {
     api_key_cmd = nil,
```
Comment (MunifTanjim) [inline README.md:56]: Why not always treat `api_key_cmd` as async and use `job:new`? 🤔

**Comment by jackMort** [comment]
As @MunifTanjin suggested, could you please just keep the api_key_cmd and use your asynchronous logic there?

**Comment by elliottminns** [comment]
> 

Absolutely. I'll make that change today!

**Review by dgagn — APPROVED** [review]
I am vouching, working very well.

**Review by jackMort — APPROVED** [review]

File: lua/chatgpt/api.lua [hunk 2]
```diff
@@ -91,22 +91,45 @@ function Api.close()
...
-        return
-      end
+      loadApiKey(Config.options.api_key_cmd)
     else
       logger.warn("OPENAI_API_KEY environment variable not set")
       return
     end
   end
-  Api.OPENAI_API_KEY = Api.OPENAI_API_KEY:gsub("%s+$", "")
+  if Api.OPENAI_API_KEY ~=nil and Api.OPENAI_API_KEY ~= "" then
```
Comment (dgagn) [inline lua/chatgpt/api.lua:130]:
```suggestion
  if Api.OPENAI_API_KEY ~= nil and Api.OPENAI_API_KEY ~= "" then
```

**Review by dgagn — COMMENTED** [review]
Just fixed the formatting error

**Pull request merged**